### PR TITLE
[Lake][Delete 1/2] support delete from lake table

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. FE creates DeletePredicatePB which is used in BE tablet metadata directly.
2. No longer use the old push task api.

### CODE REVIEW

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 276 / 399 (69.17%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|3|00.00%|[452, 453, 460]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[65, 66]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|2|3|66.67%|[43]|
|:large_blue_circle:|com/starrocks/load/DeleteJob.java|34|49|69.39%|[97, 98, 99, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 134, 181]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|125|179|69.83%|[160, 161, 163, 164, 165, 167, 179, 180, 181, 182, 184, 185, 186, 193, 194, 209, 210, 211, 234, 235, 236, 237, 238, 239, 240, 241, 242, 244, 246, 247, 249, 261, 268, 297, 301, 302, 303, 317, 318, 323, 324, 325, 326, 327, 328, 333, 339, 340, 341, 342, 344, 345, 360, 361]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|63|90|70.00%|[71, 72, 74, 75, 76, 78, 95, 119, 120, 121, 122, 124, 125, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 155]|
|:large_blue_circle:|com/starrocks/load/DeleteHandler.java|52|73|71.23%|[126, 158, 180, 186, 193, 201, 202, 203, 204, 205, 206, 208, 210, 218, 229, 372, 415, 598, 599, 600, 601]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java [hunk 1]
```diff
@@ -21,178 +21,164 @@
...
+            TransactionState state =
+                    globalTransactionMgr.getTransactionState(getDeleteInfo().getDbId(), getTransactionId());
+            if (state == null) {
+                LOG.warn("cancel delete job failed because txn not found, transactionId: {}", getTransactionId());
+            } else if (state.getTransactionStatus() == TransactionStatus.COMMITTED ||
+                    state.getTransactionStatus() == TransactionStatus.VISIBLE) {
+                LOG.warn("cancel delete job {} failed because it has been committed, transactionId: {}",
+                        getId(), getTransactionId());
+                return false;
+            } else {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java:133]: Are these check really necessary? `globalTransactionMgr.abortTransaction()` should provide the detailed error message itself.

File: fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java [hunk 2]
```diff
@@ -21,178 +21,164 @@
...
+                GlobalStateMgr.getCurrentState().getDeleteHandler()
+                        .updateTableDeleteInfo(GlobalStateMgr.getCurrentState(), db.getId(),
+                                getDeleteInfo().getTableId());
+            }
+            status = GlobalStateMgr.getCurrentGlobalTransactionMgr().
+                    getTransactionState(db.getId(), getTransactionId()).getTransactionStatus();
+        } catch (UserException e) {
+            if (cancel(DeleteHandler.CancelType.COMMIT_FAIL, e.getMessage())) {
+                throw new DdlException(e.getMessage(), e);
+            }
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java:161]: else?

File: fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java]: I add a new `TransactionAlreadyCommitException` and check it here.

File: fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/load/DeleteJob.java:155]: done

**Review by sduzh — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 276 / 398 (69.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionAlreadyCommitException.java|0|2|00.00%|[8, 9]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|0|4|00.00%|[1318, 1319, 1320, 1321]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|3|00.00%|[452, 453, 460]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[65, 66]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|2|3|66.67%|[43]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|125|179|69.83%|[160, 161, 163, 164, 165, 167, 179, 180, 181, 182, 184, 185, 186, 193, 194, 209, 210, 211, 234, 235, 236, 237, 238, 239, 240, 241, 242, 244, 246, 247, 249, 261, 268, 297, 301, 302, 303, 317, 318, 323, 324, 325, 326, 327, 328, 333, 339, 340, 341, 342, 344, 345, 360, 361]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|63|90|70.00%|[71, 72, 74, 75, 76, 78, 95, 119, 120, 121, 122, 124, 125, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 155]|
|:large_blue_circle:|com/starrocks/load/DeleteHandler.java|52|73|71.23%|[126, 158, 180, 186, 193, 201, 202, 203, 204, 205, 206, 208, 210, 218, 229, 372, 415, 598, 599, 600, 601]|
|:large_blue_circle:|com/starrocks/load/DeleteJob.java|34|42|80.95%|[98, 99, 100, 124, 125, 126, 127, 175]|

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 276 / 398 (69.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionAlreadyCommitException.java|0|2|00.00%|[8, 9]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|0|4|00.00%|[1318, 1319, 1320, 1321]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|3|00.00%|[452, 453, 460]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[65, 66]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|2|3|66.67%|[43]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|125|179|69.83%|[161, 162, 164, 165, 166, 168, 180, 181, 182, 183, 185, 186, 187, 194, 195, 210, 211, 212, 235, 236, 237, 238, 239, 240, 241, 242, 243, 245, 247, 248, 250, 262, 269, 298, 302, 303, 304, 318, 319, 324, 325, 326, 327, 328, 329, 334, 340, 341, 342, 343, 345, 346, 361, 362]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|63|90|70.00%|[72, 73, 75, 76, 77, 79, 96, 120, 121, 122, 123, 125, 126, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 156]|
|:large_blue_circle:|com/starrocks/load/DeleteHandler.java|52|73|71.23%|[126, 158, 180, 186, 193, 201, 202, 203, 204, 205, 206, 208, 210, 218, 229, 372, 415, 598, 599, 600, 601]|
|:large_blue_circle:|com/starrocks/load/DeleteJob.java|34|42|80.95%|[98, 99, 100, 124, 125, 126, 127, 175]|

**Comment by wyb** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**